js/app/application.js

```diff
@@ -141,6 +146,9 @@ const Application = new Knowledge.Class({
     vfunc_startup: function () {
         this.parent();
         let engine = Engine.get_default();
+        if (engine instanceof MoltresEngine.MoltresEngine)
```

Comment:
Maybe a nit, but might be nice to split this whole thing into a separate function `_check_for_content` with your early return here.

Would add readability, and make sure we don't have early return under not that usual circumstances that was missing some other setup we need on application startup.